Final Project PR: STAGE-REACT-API

xhundo opened the pull request.

xhundo commented:
@FiLLa-FaS Hey! I've removed the logic for signing in and signing up. Click the buttons and you'll be logged in without any credentials. You'll also be able to access the `/saved-news` route and view the page. If you have any issues with anything. Let me know!

FiLLa-FaS commented:
Hey! Thanks for sending your project for review, it already looks very good! I especially like your adaptive layout, the page looks great on intermediate screen resolutions. I left several comments below that should be fixed to get more points and pass this stage. Good luck!✨

# Summary:
### Performance Criteria

- [ ] <!--12.8--> All the blocks from the brief have been coded. The layout matches the designs for all resolutions from the mockup. Differences in size should not visually contradict the layout and get out of the grid in the layout.
    * Two bottom corners of the popup shouldn't be rounded on 320 viewport https://skr.sh/sINOaywvKiy?a
    * Header logo should be visible when the popup is opened on 320 viewport https://skr.sh/sINY1ZBEYio?a
    * Incorrect distance between `about` and footer on 768 viewport
    * All news cards should have equal height https://skr.sh/sINYjNnqV4L?a
    * Incorrect height of the news card on 768, 320 viewports
    * News cards should be centered horizontally on all screen resolutions https://skr.sh/sIN2maMMoY7?a
- [ ] <!--12.6--> The project is adapted for different screen resolutions and looks correctly on all intermediate resolutions.  There is no horizontal scrolling on resolutions of 320px and up. The scrollbar can't be hidden using the overflow: hidden declaration.
    * Your project should be centered horizontally on more than 1440px screen resolutions
- [ ] <!--12.6--> Navigation between pages and links to external resources are working correctly: there are no broken links or links leading to an empty anchor on the page and external links open in a new tab. All project pages can be accessed, and hidden blocks can be displayed.
    * All footer links should work (the last link isn't working)
    * All external links should be opened in new tab
- [ ] <!--4.2--> Classes are named according to BEM specifications.
    * Please don't use capital letters in class names https://skr.sh/sINKQS9HkbM?a (please check all project)
    * You can't use BEM element outside of its block https://skr.sh/sINk3XdCCZH?a (please check all project)
    * You shouldn't use duplicated styles for one entity https://skr.sh/sINXUfDCeUu?a (please check all project)
    * Entity can't be block and elements of this block at the same time https://skr.sh/sINPkUFizfQ?a (please check all project)
- [ ] <!--4.2--> Semantic HTML is used, meaning that semantic tags are used. All elements are used correctly (e.g. a paragraph must be a paragraph, a list must be a list). The DOM tree structure doesn't consist only of `<div />` containers.
    * `about__article` should be a paragraph
- [x] <!--4.2--> The correct approach is chosen for positioning elements, and is described using the correct syntax. For example, when positioning an element absolutely, its parent block is relatively positioned.
- [x] <!--4.2--> `flex` or `grid`  layout are used to arrange elements.
- [x] <!--4.2--> Infrastructural project files are created using CRA.
- [x] <!--4.2--> The popup windows have opening and closing functionality implemented.
- The markup has been converted into JSX:
  - [x] <!--4.2--> The markup is included within parentheses `( )`.
  - [x] <!--4.2--> The markup has been moved into the corresponding components.
- [x] <!--4.2--> Your project contains:
  - [x] An `images`
  - [x] A `components` folder with JS and CSS files for the components.
  - [x] A `fonts` folder.
- [ ] <!--4.2--> There are no linting errors or warnings.
    * https://skr.sh/sINzq8ZbWLM?a

### Best Practices

- [ ] <!--2.16--> Buttons, input fields, and links are implemented in all of the states specified in the design.
    * News card save button should have marked state https://skr.sh/sINRdn5srSN?a
- [ ] <!--2.14--> The design reuses components wherever possible.
    * You can use one component for search card and saved card https://skr.sh/sINxnFpFHYh?a
- [x] <!--2.14--> Fonts are connected using `@font-face`.
- [ ] <!--2.14--> The icons are exported in SVG format.
    * These two images should be icons https://skr.sh/sINATy8QYEz?a
- [x] <!--2.14--> Form elements should be highlighted when focused.
- [x] <!--2.14--> The form must have placeholders,  and there are required properties for the fields.
- [x] <!--2.14--> The use of `reset.css` is prohibited.


### Recommendations

- [ ] <!--1.67--> System fonts are connected as alternatives to each of your fonts.
    * You should add alternative fonts and generic families to all `font-family` styles on the project
- [ ] <!--1.67--> Images have an `alt` attribute containing appropriate values.
    * All images don't have `alt` attribute
- [x] <!--1.66--> Raster and vector images have been optimized.

## Number of points: 40

FiLLa-FaS commented:
Hey! You improved your project a lot, great job! Several small issues below are left, I hope they help you to get more points and finish the project. Good luck!✨

# Summary:
### Performance Criteria

- [ ] <!--12.8--> All the blocks from the brief have been coded. The layout matches the designs for all resolutions from the mockup. Differences in size should not visually contradict the layout and get out of the grid in the layout.
    * `newscard__topic` should be aligned at the bottom of the card https://skr.sh/sIQGQrER88P?a https://skr.sh/sIQ3ceH2KAz?a
- [x] <!--12.6--> The project is adapted for different screen resolutions and looks correctly on all intermediate resolutions.  There is no horizontal scrolling on resolutions of 320px and up. The scrollbar can't be hidden using the overflow: hidden declaration.
- [x] <!--12.6--> Navigation between pages and links to external resources are working correctly: there are no broken links or links leading to an empty anchor on the page and external links open in a new tab. All project pages can be accessed, and hidden blocks can be displayed.
- [ ] <!--4.2--> Classes are named according to BEM specifications.
    * Entity can't be block and elements of this block at the same time https://skr.sh/sIQyPybgYi8?a (please check all project)
- [x] <!--4.2--> Semantic HTML is used, meaning that semantic tags are used. All elements are used correctly (e.g. a paragraph must be a paragraph, a list must be a list). The DOM tree structure doesn't consist only of `<div />` containers.
- [x] <!--4.2--> The correct approach is chosen for positioning elements, and is described using the correct syntax. For example, when positioning an element absolutely, its parent block is relatively positioned.
- [x] <!--4.2--> `flex` or `grid`  layout are used to arrange elements.
- [x] <!--4.2--> Infrastructural project files are created using CRA.
- [x] <!--4.2--> The popup windows have opening and closing functionality implemented.
- The markup has been converted into JSX:
  - [x] <!--4.2--> The markup is included within parentheses `( )`.
  - [x] <!--4.2--> The markup has been moved into the corresponding components.
- [x] <!--4.2--> Your project contains:
  - [x] An `images`
  - [x] A `components` folder with JS and CSS files for the components.
  - [x] A `fonts` folder.
- [x] <!--4.2--> There are no linting errors or warnings.

### Best Practices

- [ ] <!--2.16--> Buttons, input fields, and links are implemented in all of the states specified in the design.
    * News card save button should have marked state https://skr.sh/sINRdn5srSN?a
- [ ] <!--2.14--> The design reuses components wherever possible.
    * You can use one component for search card and saved card https://skr.sh/sINxnFpFHYh?a
- [x] <!--2.14--> Fonts are connected using `@font-face`.
- [x] <!--2.14--> The icons are exported in SVG format.
- [x] <!--2.14--> Form elements should be highlighted when focused.
- [x] <!--2.14--> The form must have placeholders,  and there are required properties for the fields.
- [x] <!--2.14--> The use of `reset.css` is prohibited.


### Recommendations

- [ ] <!--1.67--> System fonts are connected as alternatives to each of your fonts.
    * Not all `font-family` styles have alternative fonts
- [ ] <!--1.67--> Images have an `alt` attribute containing appropriate values.
    * `savedcard-img` doesn't have `alt` attribute
- [x] <!--1.66--> Raster and vector images have been optimized.

## Number of points: 75

xhundo commented:
@FiLLa-FaS hey! I am reusing the `newscard` component in `/saved-news` route. 

Also, for the news card marked state I am using a state for it as well when it is clicked. Maybe, you can correct me if I am wrong.

xhundo commented:
@FiLLa-FaS what do you when you pointed out `header-custom__bar1` mistake? 

I would like to see where I am wrong. Is it the naming that includes `custom`? Let me know thanks!

FiLLa-FaS commented:
@xhundo 
1. It's great that you're reusing component, but I can see the card only on `saved-news` page, search and cards on the main page are not available for me. So it will be great to remove unused component from your `components` folder, so I can accept it technically without visual proof. 
2. I couldn't find marked state for the icon, but all I can see is trash icon on `saved-news` page, so I tried to do it manually. You can send me a screenshot of your card with marked state and I'll accept this criterion as well
3. BEM elements can't be used outside of its class. It means that I can't use `header-bar__item` class if there's no `header-bar` BEM block as a wrapper of this element. There's no `header-custom` class in the project, so you can't use elements of this unexisted block. You have `header` block, so you can use elements like `header__abcd` inside `header`. You can learn more here https://en.bem.info/methodology/quick-start/#element

xhundo commented:
@FiLLa-FaS Thanks for the feedback, I really appreciate it! I've corrected those mistakes. Also, below is the screenshot with my `newscard` with marked state. I've removed the `savedcard` component. You should see my `newscard` reusable component in `/saved-news` route. 

![markedcards](https://user-images.githubusercontent.com/98115017/219486248-e6e4060b-f5b4-4ffb-80e1-be6899ae7dd7.jpg)

FiLLa-FaS commented:
Hey! I'm glad to say that your project has been approved, and i really appreciate your efforts. You still can fix last two issues (you fixed `header__bar` elements correctly, but there're more similar cases in the project) to improve your project if you want, but now you can move on. Good luck on the next diploma stage!

# Summary:
### Performance Criteria

- [x] <!--12.8--> All the blocks from the brief have been coded. The layout matches the designs for all resolutions from the mockup. Differences in size should not visually contradict the layout and get out of the grid in the layout.
- [x] <!--12.6--> The project is adapted for different screen resolutions and looks correctly on all intermediate resolutions.  There is no horizontal scrolling on resolutions of 320px and up. The scrollbar can't be hidden using the overflow: hidden declaration.
- [x] <!--12.6--> Navigation between pages and links to external resources are working correctly: there are no broken links or links leading to an empty anchor on the page and external links open in a new tab. All project pages can be accessed, and hidden blocks can be displayed.
- [ ] <!--4.2--> Classes are named according to BEM specifications.
    * BEM element can't be used outside of its block https://skr.sh/sIRWTt0ix5U?a (you can see the example on the screenshot, but you should fix all similar cases on both pages throughout the project)
- [x] <!--4.2--> Semantic HTML is used, meaning that semantic tags are used. All elements are used correctly (e.g. a paragraph must be a paragraph, a list must be a list). The DOM tree structure doesn't consist only of `<div />` containers.
- [x] <!--4.2--> The correct approach is chosen for positioning elements, and is described using the correct syntax. For example, when positioning an element absolutely, its parent block is relatively positioned.
- [x] <!--4.2--> `flex` or `grid`  layout are used to arrange elements.
- [x] <!--4.2--> Infrastructural project files are created using CRA.
- [x] <!--4.2--> The popup windows have opening and closing functionality implemented.
- The markup has been converted into JSX:
  - [x] <!--4.2--> The markup is included within parentheses `( )`.
  - [x] <!--4.2--> The markup has been moved into the corresponding components.
- [x] <!--4.2--> Your project contains:
  - [x] An `images`
  - [x] A `components` folder with JS and CSS files for the components.
  - [x] A `fonts` folder.
- [x] <!--4.2--> There are no linting errors or warnings.

### Best Practices

- [x] <!--2.16--> Buttons, input fields, and links are implemented in all of the states specified in the design.
- [x] <!--2.14--> The design reuses components wherever possible.
- [x] <!--2.14--> Fonts are connected using `@font-face`.
- [x] <!--2.14--> The icons are exported in SVG format.
- [x] <!--2.14--> Form elements should be highlighted when focused.
- [x] <!--2.14--> The form must have placeholders,  and there are required properties for the fields.
- [x] <!--2.14--> The use of `reset.css` is prohibited.


### Recommendations

- [ ] <!--1.67--> System fonts are connected as alternatives to each of your fonts.
    * https://skr.sh/sIR56IW4Ne6?a https://skr.sh/sIR9L86oIuN?a
- [x] <!--1.67--> Images have an `alt` attribute containing appropriate values.
- [x] <!--1.66--> Raster and vector images have been optimized.

## Number of points: 94
